pouchdb-mapreduce/index.d.ts

```diff
@@ -17,6 +55,12 @@ declare namespace PouchDB {
          */
         viewCleanup(callback: PouchDB.Core.Callback<any,void>): void;
         viewCleanup(): Promise<void>;
+
+        query(fun: string, opts: Query.Options, callback: (err: Core.Error, res: Query.Response) => void): void;
```

Comment:
Any reason why You have so many overrides here? For callback style you can use `fun: string | Filter | Function` and at least reduce it to two overrides.
Also please add Doc info comment for each line.